src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseAPI.php

```diff
@@ -0,0 +1,426 @@
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\InvalidParamException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\ClaimExpense;
+use OrangeHRM\ORM\Exception\TransactionException;
+
+class ClaimExpenseAPI extends Endpoint implements CrudEndpoint
+{
+    use EntityManagerHelperTrait;
+    use ClaimServiceTrait;
+    use DateTimeHelperTrait;
+    use AuthUserTrait;
+    use UserRoleManagerTrait;
+    use UserServiceTrait;
+
+    public const PARAMETER_EXPENSE_TYPE_ID = 'expenseTypeId';
+    public const PARAMETER_AMOUNT = 'amount';
+    public const PARAMETER_NOTE = 'note';
+    public const PARAMETER_DATE = 'date';
+    public const PARAMETER_REQUEST_ID = 'requestId';
+    public const PARAMETER_CLAIM_REQUEST_ID = 'claimRequestId';
+    public const NOTE_MAX_LENGTH = 1000;
+    public const PARAMETER_TOTAL_AMOUNT = 'totalAmount';
+
+    /**
+     * @OA\Get(
+     *     path="/api/v2/claim/expenses",
+     *     tags={"Claim/Expenses"},
+     *     @OA\Parameter(
+     *         name="claimRequestId",
+     *         in="query",
+     *         required=false,
+     *     )
+     *     @OA\Response(
+     *         response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 type="array",
+     *                 @OA\Items(ref="#/components/schemas/Claim-ExpenseModel")
+     *             ),
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $requestId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_CLAIM_REQUEST_ID);
+        if ($this->getClaimService()->getClaimDao()->getClaimRequestById($requestId) == null) {
+            throw new InvalidParamException();
+        }
+        $claimExpenses = $this->getClaimService()->getClaimDao()->getClaimExpenseList($requestId);
+        $count = $this->getClaimService()->getClaimDao()->getClaimExpenseCount($requestId);
+        $total = $this->getClaimService()->getClaimDao()->getClaimExpenseTotal($requestId);
```

Comment:
seems wrong. All getAll APIs should handle pagination, without considering whether FE need it